src/components/config/OidcConfig.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+import { AuthProvider, User } from 'oidc-react';
+import {
+    authority,
+    clientId,
+    configResponseKey,
+    cspRoleKey,
+    currentUserKey,
+    redirectUri,
+    responseType,
+    scope,
+    userRolesKey,
+} from './config';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { Location } from '@remix-run/router';
+import { homePageRoute, isAuthenticatedKey, usernameKey } from '../utils/constants';
+
+export const OidcConfig = (): JSX.Element => {
+    const navigate = useNavigate();
+    const location: Location = useLocation();
+    const oidcConfig = {
+        // eslint-disable-next-line @typescript-eslint/require-await
+        onSignIn: async (response: User | null) => {
+            if (response === null) {
+                return;
+            }
+            localStorage.setItem(configResponseKey, response.toStorageString());
+            localStorage.setItem(currentUserKey, response.profile.name === undefined ? '' : response.profile.name);
+            const roles: unknown = response.profile['urn:zitadel:iam:org:project:roles'];
+            if (Array.isArray(roles)) {
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+                const objectKeys = Object.keys(roles[0]);
+                localStorage.setItem(isAuthenticatedKey, 'true');
```

Comment:
we must avoid setting these values. 
Instead use the 'useAuth' hook to get the values.